crates/yttrium/src/chain_abstraction/client.rs

```diff
@@ -239,23 +242,55 @@ impl Client {
             ));
         }
         let estimated_initial_transaction = estimate_gas_fees(
-            route_response.initial_transaction.clone(),
+            prepare_response.initial_transaction.clone(),
             &eip1559_fees,
             initial_l1_data_fee,
         );
 
         Ok(ui_fields::ui_fields(
-            route_response,
+            prepare_response,
             estimated_transactions,
             estimated_initial_transaction,
             fungibles,
         ))
     }
 
+    // TODO test
+    pub async fn prepare_detailed(
```

Comment:
What does it do? Can WK call it instead of prepare?

Reply:
> Can WK call it instead of prepare?

Yes that's the goal. We might even consider renaming `prepare_detailed` to `prepare` and `prepare` to something else e.g. `prepare_step1`